src/game.cpp

```diff
@@ -14561,11 +14561,11 @@ void game::start_calendar()
         if( scen->has_flag("SPR_START") ) {
             ; // Do nothing;
         } else if( scen->has_flag("SUM_START") ) {
-            calendar::start += DAYS((int)ACTIVE_WORLD_OPTIONS["SEASON_LENGTH"]);
+            calendar::start += DAYS( calendar::season_length() );
```

Comment:
These changes (x6) don't look consistent. The function's name is `start_calendar()` so it's supposed to initialize it, not rely on it's values. There are alot of `ACTIVE_WORLD_OPTIONS` around whilst `calendar::set_season_length` is invoked from fairly weird places (`options_manager::serialize` and `worldfactory::set_active_world`). It should probably be placed here and be initialized from `ACTIVE_WORLD_OPTIONS` too.